tests/testing_support/mock_external_openai_server.py

```diff
@@ -0,0 +1,117 @@
+# Copyright 2010 New Relic, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+
+from testing_support.mock_external_http_server import MockExternalHTTPServer
+
+
+# This defines an external server test apps can make requests to instead of
+# the real OpenAI backend. This provides 3 features:
+#
+# 1) This removes dependencies on external websites.
+# 2) Provides a better mechanism for making an external call in a test app than
+#    simple calling another endpoint the test app makes available because this
+#    server will not be instrumented meaning we don't have to sort through
+#    transactions to separate the ones created in the test app and the ones
+#    created by an external call.
+# 3) This app runs on a separate thread meaning it won't block the test app.
+
+ALLOWED_PATHS = set(["/chat/completions"])
```

Comment:
We will also need to test a fake path for errors and the embeddings endpoint as well. Could that be added to this?

Reply:
Honestly could probably drop this check altogether, it doesn't feel as necessary after I used the SDK enough.

Reply:
That's fair - we just need a mock response in here for the embeddings response object

Reply:
I've updated the embedding tests as well and ensured they work with the mocking setup.